### ISSUE TITLE
Exception 'numHits must be > 0' raised when subquery request used

### ISSUE THREAD

**Issue by shm-dmitry** [crate/crate#9354]
**CrateDB version**: 4.0.8

This error always reproduced on version 4.0.8. 
At 4.0.7 this query works fine.

**Problem description**:
I receive error
`ERROR: SQLParseException: numHits must be > 0; please use TotalHitCountCollector if you just need the total hit count`

when executing request like 
```
SELECT
    (
        CASE
            WHEN c.id IN(
                SELECT
                    catalog_id
                FROM
                    catalog_favorites
                WHERE
                    user_id = '01-00000000000000019835'
            )THEN
                1
            ELSE
                0
        END
    ) AS favorite
FROM
    catalog c
limit 5;
```

`catalog_favorites` table:
```
CREATE TABLE IF NOT EXISTS "doc"."catalog_favorites" (
   "added" TIMESTAMP WITH TIME ZONE,
   "catalog_id" TEXT,
   "is_active" BOOLEAN,
   "source" TEXT,
   "user_id" TEXT,
   "user_order" INTEGER,
   PRIMARY KEY ("catalog_id", "user_id", "source")
)
```

I found, that problem appear in WHERE statement of subquery (it will disappear when I change where statement).
Without subquery all works fine.
Without WHERE in subquery all works fine.
TCP dump in attach (rename .txt into .pcap).
[dump.pcap.txt](https://github.com/crate/crate/files/3858097/dump.pcap.txt)

Exception raised from TopFieldCollector.java:393 (create method)

**Steps to reproduce**:

Sorry, I cant create plain example - error does not reproduced on samples.
I will try to it do again some hours later.

**Comment by mfussenegger** [comment]
This might be a regression introduced by https://github.com/crate/crate/commit/3c6e602a1d, we'll have a closer look.

Thank you for the bug report.

**Comment by mfussenegger** [comment]
We've a fix for this which will be released with 4.0.9 (https://github.com/crate/crate/pull/9359)

Thank you again for bringing this up.
